feature_layers/dynamic-entity-layer/src/main/java/com/esri/samples/dynamic_entity_layer/DynamicEntityLayerSample.java

```diff
@@ -0,0 +1,307 @@
+
+      // set the viewpoint to the map view
+      mapView.setViewpoint(new Viewpoint(utahSandyEnvelope));
+
+      // create a graphics overlay and add a graphic to show the envelope's extent using a simple line symbol
+      var borderLineSymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.DASH, Color.RED, 2);
+      var graphicsOverlay = new GraphicsOverlay();
+      var graphic = new Graphic(utahSandyEnvelope, borderLineSymbol);
+      graphicsOverlay.getGraphics().add(graphic);
+
+      // add the graphics overlay to the map view
+      mapView.getGraphicsOverlays().add(graphicsOverlay);
+
+      // create a controls vbox containing the UI components
+      setUpControlsVBox();
+
+      // create the stream service as a dynamic entity data source
+      var streamServiceUrl = "https://realtimegis2016.esri.com:6443/arcgis/rest/services/SandyVehicles/StreamServer";
+      var streamService = new ArcGISStreamService(streamServiceUrl);
+
+      // add a filter for the data source to limit the amount of data received by the application
+      var streamServiceFilter = new ArcGISStreamServiceFilter();
+      streamServiceFilter.setGeometry(utahSandyEnvelope);
+      streamServiceFilter.setWhereClause("speed > 0");
+      streamService.setFilter(streamServiceFilter);
+
+      // set a duration of five minutes for how long observation data is stored in the data source
+      streamService.getPurgeOptions().setMaximumDuration(5);
+
+      // create a layer to display the data from the stream service
+      dynamicEntityLayer = new DynamicEntityLayer(streamService);
+
+      // create unique value renderers for customizing the dynamic entity layer appearance
+      var uniqueValueRenderers = createUniqueRenderers();
+
+      // set the dynamic entity and previous observation renderers
+      dynamicEntityLayer.setRenderer(uniqueValueRenderers.get(0));
+      dynamicEntityLayer.getTrackDisplayProperties().setPreviousObservationRenderer(uniqueValueRenderers.get(1));
+
+      // create a simple line renderer and set the track line renderer to it
+      var trackLineRenderer = new SimpleRenderer(new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.LIGHTGRAY, 2));
+      dynamicEntityLayer.getTrackDisplayProperties().setTrackLineRenderer(trackLineRenderer);
+
+      // update the label with the service connection status
+      connectionStatusLabel.textProperty().bind(Bindings.createStringBinding(() ->
+        "Status: " + streamService.connectionStatusProperty().asString().get(), streamService.connectionStatusProperty()));
+
+      // update the button text with the service connection status
+      connectionButton.textProperty().bind(Bindings.createStringBinding(() ->
+        streamService.getConnectionStatus() == ConnectionStatus.CONNECTED ?
+          "Disconnect" : "Connect", streamService.connectionStatusProperty()));
+
+      // toggle the service connection on button press
+      connectionButton.setOnAction(a -> {
```

Comment:
Something to explore - it looks like there's an onActionProperty, can we use that here?

Reply:
I don't see a straightforward property binding that we can do here and using the property listener might make things a bit more complicated (?).